pubsub/acks_test.go

```diff
@@ -342,3 +342,86 @@ func TestReceiveReturnsErrorFromSendAcks(t *testing.T) {
 		}
 	}
 }
+
+// callbackDriverSub implements driver.Subscription and allows something like
+// monkey patching of both its ReceiveBatch and SendAcks methods.
+type callbackDriverSub struct {
```

Comment:
Does this generalize some of the other test subs?

Reply:
It could probably generalize all the other testing sub drivers. Not sure it's worth the effort though.